spec/models/day_spec.rb

```diff
@@ -1,43 +1,44 @@
 require 'rails_helper'
 
 RSpec.describe Day, type: :model do
-  fixtures :all
+  fixtures :days, :students, :attendances
 
   describe '#has_student?' do # # for an instance method, . for a class method
-    context 'day has student' do
-      let(:student) { Student.new }
+    context 'day has this student' do
+      let(:student) { Student.new } # why we didn't use fixture here??? like :student1
 
       subject(:day) do
-        described_class.new(students: [student])
+        Day.new(students: [student])
       end
 
       it 'returns true' do
         expect(subject.has_student?(student)).to be true
       end
     end
-
+# so we let things which we need but aren't the object of the test and use
+# subject when we refer to an object under test?
     context 'day doesnt have this student' do
-      let(:student) { Student.new }
       let(:student1) { Student.new }
+      let(:student2) { Student.new }
 
       subject(:day) do
-        described_class.new(students: [student])
+        Day.new(students: [student1])
       end
 
       it 'returns false' do
-        expect(subject.has_student?(student1)).to be false
+        expect(day.has_student?(student2)).to be false
       end
     end
   end
 
   describe '#students_without_attendance' do
     context 'day has no attendances' do
-      subject(:day) do
-        described_class.new
+      subject(:day) do # is this equivalent to let(:day) { Day.new } ???
```

Comment:
Yes, they're almost equivalent. The only difference would be that if you use with `let` you cannot use `subject` in the `it` block (see example and stackoverflow link above).

Reply:
Yep, but I probably wouldn’t do it anyway, cause it’s less clear